### PR TITLE
HPCC-22308 Make fieldTranslation options consistent

### PR DESCRIPTION
**Pull request by jakesmith**
Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by jakesmith** [comment]
@ghalliday please review

**Review by ghalliday — COMMENTED** [review]
@jakesmith looks ok.  One comment

File: initfiles/componentfiles/configschema/xsd/thor.xsd [hunk 1]
```diff
@@ -223,7 +223,17 @@
                               hpcc:tooltip="HTTP Header field to use for sending and receiving CallerId"/>
                 <xs:attribute name="httpGlobalIdHeader" type="xs:string" hpcc:displayName="HTTP GlobalId Header" hpcc:presetValue="HPCC-Global-Id"
                               hpcc:tooltip="HTTP Header field to use for sending and receiving GlobalId"/>
-
+                <xs:attribute name="fieldTranslationEnabled" hpcc:displayName="Enable Field Translation"
+                              hpcc:presetValue="false"
```
[the remark this answers is not in the document]
Comment (jakesmith) [reply initfiles/componentfiles/configschema/xsd/thor.xsd]: Yes it probably should, it does in componentfiles/configxml/ xsd's

This (configschema) xsd seting was a copy&paste from same existing section in componentfiles/configschema/roxie.xsd which also has "false".

And noticed that this setting is missing in the eclagent xsd under configschema's as well.

I'll add missing property to componentfiles/configschema/eclagent.xsd and change all to "payload"

**Comment by jakesmith** [comment]
@ghalliday - see comments above and additional commit.

**Comment by ghalliday** [comment]
@jakesmith roxie still uses the old flag - e.g., ccdquery.cpp(374):

    updateFromWorkUnit(enableFieldTranslation, wu, "layoutTranslation");  // Name is different for compatibility reasons

**Comment by jakesmith** [comment]
@ghalliday - all engines still use "layoutTranslation" as a workunit option (or as a HINT). And now (with this PR) "fieldTranslationEnabled" as global config option.

Roxie and Hthor used "fieldTranslationEnabled" as a global config option only. Thor used to (prior to this PR), use "layoutTranslation" at all levels.

I was in 2 minds whether to change Roxie and Hthor to match Thor (i.e. everywhere use "layoutTranslation" and deprecate "fieldTranslationEnabled" altogether,
or to change Thor to match Hthor/Roxie, i.e. "fieldTranslationEnabled" as global config option, and "layoutTranslation" at workunit/activity hint option.

In the end I decided to change Thor because Thor did not have a xsd setting and Roxie/HThor had their the pre-existing "fieldTranslationEnabled" options.
And partly because of that comment in Roxie (// Name is different for compatibility reasons) - suggested it wasn't changed to match before for a reason.

**Comment by ghalliday** [comment]
@jakesmith ok, that makes sense.  I was confused by all the changes to the ecl files in ecl/regress.  I suspect those changes should be regressed - even though those queries are generally not run.

**Comment by jakesmith** [comment]
Oh it was more inconsistent than my last message made out.

Although eclagent had/generated the configuration (xsd) option "fieldTranslationEnabled".
It never used it, but instead had it's own option (3rd variety) in code (and in the regress/ecl tests) "layoutTranslationEnabled".

I changed "layoutTranslationEnabled" in the regress/ecl tests, but should have changed it to the workunit option "layoutTranslation" not the global config option "fieldTranslationEnabled"
Will fix those.

**Comment by jakesmith** [comment]
@ghalliday - please see ecl/regress changes.

**Comment by ghalliday** [comment]
@jakesmith looks good, please can you squash.

@richardkchapman please sanity check and merge

**Comment by jakesmith** [comment]
Squashed.

**Pull request merged**